src/frontends/onnx/frontend/src/op/trilu.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/trilu.hpp"
+
+#include "default_opset.hpp"
+#include "exceptions.hpp"
+#include "onnx_import/core/null_node.hpp"
+
+namespace ngraph {
+namespace onnx_import {
+namespace op {
+namespace set_14 {
+
+OutputVector trilu(const Node& node) {
+    const auto inputs = node.get_ng_inputs();
+    const auto num_inputs = inputs.size();
+
+    CHECK_VALID_NODE(node, num_inputs > 0 && num_inputs <= 2, "Trilu expects <= 2 input tensors. Got: ", num_inputs);
+
+    const auto& input = inputs[0];
+    const auto& rank = input.get_partial_shape().rank();
+    if (rank.is_static()) {
+        CHECK_VALID_NODE(node, rank.get_length() >= 2, "Trilu first input's rank must be >= 2");
+    }
+    bool is_k_available = num_inputs == 2 && !ngraph::op::is_null(inputs[1]);
+    if (is_k_available) {
+        CHECK_VALID_NODE(node, inputs[1].get_partial_shape().compatible({}), "Trilu second input must be a scalar");
+    }
+
+    const auto shape = std::make_shared<default_opset::ShapeOf>(input);
+    const auto zero = default_opset::Constant::create(element::i64, Shape{}, {0});
+    const auto one = default_opset::Constant::create(element::i64, Shape{}, {1});
+
+    // The approach here is to create a mask, that later can be used in Select operator
+    // to choose appropiate values from the input
+    //
+    // For example N = 4, M = 5, k = -1, upper = false
+    // horizontal_range = [[0, 1, 2, 3, 4]]
+    // vertical_range = [[-1],
+    //                   [0],
+    //                   [1],
+    //                   [2]]
+    // Since upper == false, we compare horizontal_range <= vertical_range
+    // and thanks to broadcasting, it conceptually looks like:
+    // [[0, 1, 2, 3, 4],        [[-1, -1, -1, -1, -1],
+    //  [0, 1, 2, 3, 4],   <=    [0, 0, 0, 0, 0],
+    //  [0, 1, 2, 3, 4],         [1, 1, 1, 1, 1],
+    //  [0, 1, 2, 3, 4]]         [2, 2, 2, 2, 2]]
+    //
+    // which results in following mask:
+    // [[0, 0, 0, 0, 0],
+    //  [1, 0, 0, 0, 0],
+    //  [1, 1, 0, 0, 0],
+    //  [1, 1, 1, 0, 0]]
+    //
+    // That matches ONNX spec: "If upper is set to false, ... a negative k value excludes the main diagonal
+    // and (|k|-1) diagonals below it"
+
+    // fetch last two dimensions of input shape
+    // M = shape[-1]
+    // N = shape[-2]
+    const auto M = std::make_shared<default_opset::Gather>(shape,
+                                                           default_opset::Constant::create(element::i32, Shape{}, {-1}),
+                                                           zero);
+    const auto N = std::make_shared<default_opset::Gather>(shape,
+                                                           default_opset::Constant::create(element::i32, Shape{}, {-2}),
+                                                           zero);
```

Comment:
may it is better to use `Slice` operation for dimension retrieving because FWs use either `StridedSlice` or `Slice` and `Gather` behavior for negative indices is changing from one opset to the other.